lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -2231,52 +3323,66 @@ class DomSegment {}
 
 extension DomSegmentExtension on DomSegment {
   @JS('index')
-  external double get _index;
-  int get index => _index.toInt();
+  external JSNumber get _index;
+  int get index => _index.toDart.toInt();
+
+  @JS('isWordLike')
+  external JSBoolean get _isWordLike;
+  bool get isWordLike => _isWordLike.toDart;
+
+  @JS('segment')
+  external JSString get _segment;
+  String get segment => _segment.toDart;
 
-  external bool get isWordLike;
-  external String get segment;
-  external String get breakType;
+  @JS('breakType')
+  external JSString get _breakType;
+  String get breakType => _breakType.toDart;
 }
 
 DomSegmenter createIntlSegmenter({required String granularity}) {
-  final Object? segmenterConstructor = domIntl.Segmenter;
-  if (segmenterConstructor == null) {
+  if (domIntl.Segmenter == null) {
     throw UnimplementedError('Intl.Segmenter() is not supported.');
   }
 
-  return js_util.callConstructor<DomSegmenter>(
-    segmenterConstructor,
-    <Object?>[
-      <String>[],
-      js_util.jsify(<String, String>{'granularity': granularity}),
-    ],
+  return DomSegmenter(
+    <JSAny?>[].toJS,
```

Comment:
This can also be `JSUndefined`, if that's possible here. I switched it to an empty array previously because there was no way for me to pass `JSUndefined` into the constructor without it round-tripping through and becoming null. If `JSUndefined` would be `undefined` all the way down, I think it would be better to pass that than an empty array.

Reply:
Okay, `JSUndefined` isn't reified yet unfortunately. This is for compatibility reasons with how JS backends handle `null` / `undefined`. Let me add a comment to revisit this when we reify `JSUndefined` on Dart2Wasm.